src/Firebase/AppCheck/AppCheckTokenVerifier.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase\AppCheck;
+
+use Firebase\JWT\CachedKeySet;
+use Firebase\JWT\JWT;
+use Kreait\Firebase\Exception\AppCheck\FailedToVerifyAppCheckToken;
+use Kreait\Firebase\Exception\AppCheck\InvalidAppCheckToken;
+use LogicException;
+use UnexpectedValueException;
+
+use function in_array;
+use function is_string;
+use function str_starts_with;
+
+/**
+ * @internal
+ */
+class AppCheckTokenVerifier
+{
+    private const APP_CHECK_ISSUER_PREFIX = 'https://firebaseappcheck.googleapis.com/';
+    private string $projectId;
+    private CachedKeySet $keySet;
+
+    public function __construct(string $projectId, CachedKeySet $keySet)
```

Comment:
```suggestion
    /**
     * @param non-empty-string $projectId
     */
    public function __construct(string $projectId, CachedKeySet $keySet)
```

Annotations like this in general could be added or improved, but don't invest too much time into it, just when you are working at those locations anyway 🙏